Fix import server files dir traversal

TobiSW opened the pull request:
<!--

Before working on a contribution, you must determine on which branch you need to work:
- Bug fix: choose the latest maintenance branch `10.5`
- Feature/Improvement: choose `11.x` 

> All bug fixes merged into the latest maintenance branch are also merged to the current dev branch (`11.x`) on a regular basis.

## Please make sure your PR complies with all of the following points: 
- [ ] Read and accept our [contributing guidelines](/CONTRIBUTING.md) before you submit a PR.
- [ ] Features need to be proper documented in `doc/` 
- [ ] Bugfixes need a short guide how to reproduce them -> target branch is the oldest supported maintenance branch, e.g. `10.5` (see Readme.md for the list of supported versions)
- [ ] Meet all coding standards (see PhpStan actions) 

**Don't submit a PR if it doesn't comply, it'll be closed without a comment!**
-->  
  

## Changes in this pull request  
Fix import server files action to prevent directory traversal

## Additional info

### Request
POST /admin/asset/import-server-files HTTP/2
[...]
parentId=1&serverPath=/var/log&files=/../../../../../etc/passwd

### Response
HTTP/2 200 OK
[...]
{"success":true}

TobiSW committed:
Fix import server files action to prevent directory traversal

TobiSW committed:
Fix import server files action to prevent directory traversal

TobiSW committed:
Fix import server files action to prevent directory traversal

- adjust code comment

robertSt7 commented on bundles/AdminBundle/Controller/Admin/Asset/AssetController.php:
```diff
@@ -2477,6 +2477,14 @@ public function importServerFilesAction(Request $request)
+        // prevent directory traversal
+        foreach ($files as $file) {
+            $absolutePath = realpath($serverPath.$file);
+            if (!str_starts_with($absolutePath, rtrim(str_replace('../', '', $serverPath), './'))) {
```
```suggestion
            if (!str_starts_with($absolutePath, $serverPath)) {
```
@TobiSW  I think we don't need the string manipulation for `$serverPath`, because the realpath is stored in the variable https://github.com/pimcore/pimcore/pull/15664/files#diff-8a94426359af1cadbcb7fdeb7af5731ce2ac295a758ed07a38d36db9d5d85ff9R2473 
What do you think?

TobiSW replied:
@robertSt7 thx for the review and your feedback, you are right, i removed the string manipulation check

TobiSW committed:
Fix import server files action to prevent directory traversal

- remove not needed string manipulation check of serverPath as it is already checked before

robertSt7 commented:
@TobiSW Thanks a lot for that PR :)
